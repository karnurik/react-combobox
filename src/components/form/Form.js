import React, { Component } from 'react';
import Combobox from '../common/combobox/Combobox'
import './form.css';

class Form extends Component {
    constructor(props) {
        super(props);
        this.state={
            options : [
                { id: '1', name: 'Candy Eat' },
                { id: '2', name: 'Onions Pick' },
                { id: '3', name: 'Fanny Pick' },
                { id: '4', name: 'Click o Wisp' }],
            userName : ''
        }
    }

    handleInputChange = (e) => {
        const {value} = e.target;
        this.setState({userName: value});
    }

    render() {
        return (
            <>
                <form  className="form">
                    <label>Username</label>
                    <input type="text" name="username" placeholder="Username" value={this.state.userName} onChange={this.handleInputChange} />

                    <Combobox
                        data={this.state.options}
                        placeholder='Select Games'
                        searchPlaceholder='Search Games'
                        label='Games'
                        />
                
                    <input type="submit" value="Submit" />
                </form>
            </>
        );
    }
}

export default Form;