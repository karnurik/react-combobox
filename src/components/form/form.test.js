import React from 'react';
import Form from './form';
import renderer from 'react-test-renderer';


const options =  [
    { id: '1', name: 'Candy Eat' },
    { id: '2', name: 'Onions Pick' },
    { id: '3', name: 'Fanny Pick' },
    { id: '4', name: 'Click o Wisp' }
];

const userName = 'UjjwalKumar';

test('Component renders successfully', () => {
  const component = renderer.create(
    <Form
      options={options}
      userName={userName}
    />
   );
  
  expect(component.toJSON()).toMatchSnapshot();
});