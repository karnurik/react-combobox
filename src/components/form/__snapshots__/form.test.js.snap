// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders successfully 1`] = `
<form
  className="form"
>
  <label>
    Username
  </label>
  <input
    name="username"
    onChange={[Function]}
    placeholder="Username"
    required={true}
    type="text"
    value=""
  />
  <div
    className="combo-box"
  >
    <label>
      Games
    </label>
    <input
      className="form-input"
      defaultValue=""
      onFocus={[Function]}
      placeholder="Select Games"
      type="text"
    />
  </div>
  <input
    disabled={true}
    type="submit"
    value="Submit"
  />
</form>
`;
